streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStreamThread.java

```diff
@@ -311,6 +314,8 @@ public void run() {
                 "Updating global state failed. You can restart KafkaStreams to recover from this error.",
                 recoverableException
```

Comment:
Hm ok this might be a problem. Since this is thrown from another catch block and not from the try block, it won't be caught by the catch block below and will slip through the exception handler.

Reply:
like in stream thread we can just add a call to the handler